src/logic/ratings.tsx

```diff
@@ -0,0 +1,204 @@
+// @flow
+import { useEffect, useCallback } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { useNavigation } from "@react-navigation/native";
+import { add, isBefore, parseISO } from "date-fns";
+import AsyncStorage from "@react-native-async-storage/async-storage";
+import useFeature from "@ledgerhq/live-common/lib/featureFlags/useFeature";
+import { accountsCountSelector } from "../reducers/accounts";
+import {
+  ratingsModalOpenSelector,
+  ratingsCurrentRouteSelector,
+  ratingsHappyMomentSelector,
+  ratingsDataOfUserSelector,
+} from "../reducers/ratings";
+import {
+  setRatingsModalOpen,
+  setRatingsCurrentRoute,
+  setRatingsHappyMoment,
+} from "../actions/ratings";
+import { languageSelector } from "../reducers/settings";
+import { track } from "../analytics";
+
+const ratingsDataOfUserAsyncStorageKey = "ratingsDataOfUser";
+
+const getCurrentRoute = (
+  state: NavigationState | Required<NavigationState["routes"][0]>["state"],
+): Routes | undefined => {
+  if (state.index === undefined || state.index < 0) {
+    return undefined;
+  }
+  const nestedState = state.routes[state.index].state;
+  if (nestedState !== undefined) {
+    return getCurrentRoute(nestedState);
+  }
+  return state.routes[state.index].name;
+};
+
+export async function getRatingsDataOfUserFromStorage() {
+  const ratingsDataOfUser = await AsyncStorage.getItem(
+    ratingsDataOfUserAsyncStorageKey,
+  );
+
+  return JSON.parse(ratingsDataOfUser);
+}
+
+export async function setRatingsDataOfUserInStorage(ratingsDataOfUser) {
+  await AsyncStorage.setItem(
+    ratingsDataOfUserAsyncStorageKey,
+    JSON.stringify(ratingsDataOfUser),
+  );
+}
+
+const useRatings = () => {
+  const ratings = useFeature("ratings");
```

Comment:
maybe rename this to `ratingsFeature` or `ratingsConfig` ?
Also it would be great to have a typing more specific than `any` as this object seems to contain a variety of pretty different stuff